src/sql/src/plan.rs

```diff
@@ -483,11 +485,15 @@ impl Params {
 #[derive(Ord, PartialOrd, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash, Copy)]
 pub struct PlanContext {
     pub wall_time: DateTime<Utc>,
+    pub qgm_optimizations: bool,
 }
 
 impl PlanContext {
-    pub fn new(wall_time: DateTime<Utc>) -> Self {
-        Self { wall_time }
+    pub fn new(wall_time: DateTime<Utc>, qgm_optimizations: bool) -> Self {
```

Comment:
There is already a datastructure that provisions for more options - `OptimizerConfig`, which can be derived from a `StatementContext` that wraps a `PlanContext`. Going forward we will probably populate the `OptimizerConfig` from CLI flags rather than session options, so the `PlanContext` field will most likely disappear.